### PR TITLE
Map tls.ClientAuthType to s2av2 proto RequestClientCertificate

### PR DESCRIPTION
**Pull request by rmehta19**
This PR is a fix for the issue described in https://github.com/google/s2a-go/pull/43 's description.

Update s2a.proto RequestClientCertificate enum to match [tls.ClientAuthType](https://pkg.go.dev/crypto/tls#ClientAuthType)

Rerun //tools/proto/regenerate_proto.sh to pick up the new changes, and update:  
//internal/proto/v2/s2a_pb_go/s2a.pb.go  
//internal/proto/v2/s2a_pb_go/s2a_grpc.pb.go  

The script also updated a bunch of other proto generated files throughout this repo.

### CODE REVIEW

File: internal/v2/tls_config_store/tls_config_store.go [hunk 1]
```diff
@@ -168,32 +168,52 @@ func GetTlsConfigurationForServer(cstream s2av2pb.S2AService_SetUpSessionClient,
...
+	case s2av2pb.GetTlsConfigurationResp_ServerTlsConfiguration_REQUEST_AND_REQUIRE_CLIENT_CERTIFICATE_AND_VERIFY:
+		// This case actually maps to tls.RequireAndVerifyClientCert. However this
+		// mapping triggers normal verification, followed by custom verification,
+		// specified in VerifyPeerCertificate. To bypass normal verification, and
+		// only do custom verification we set clientAuth to RequireAnyClientCert or
+		// RequestClientCert. See https://github.com/google/s2a-go/pull/43 for full
+		// discussion.
+		clientAuth = tls.RequireAnyClientCert
+	default:
+		// TODO(rmehta19): consider what the default should be.
```
Comment (matthewstevenson88) [inline internal/v2/tls_config_store/tls_config_store.go:212]: I think tls.RequireAnyClientCert should be the default. :)

File: internal/v2/tls_config_store/tls_config_store.go [hunk 2]
```diff
@@ -168,32 +168,52 @@ func GetTlsConfigurationForServer(cstream s2av2pb.S2AService_SetUpSessionClient,
...
-		// TODO(rmehta19): Remove "+ 2" when proto file enum change is merged.
-		ClientAuth: tls.ClientAuthType(tlsConfig.RequestClientCertificate) + 2,
+		ClientAuth: clientAuth,
 		MinVersion: minVersion,
 		MaxVersion: maxVersion,
 	}, nil
 }
 
+
+func getTLSClientAuthType(tlsConfig *s2av2pb.GetTlsConfigurationResp_ServerTlsConfiguration) tls.ClientAuthType {
```
[the remark this answers is not in the document]
Comment (rmehta19) [reply internal/v2/tls_config_store/tls_config_store.go:186]: Will do, thanks!

File: internal/v2/tls_config_store/tls_config_store.go [hunk 3]
```diff
@@ -168,32 +168,52 @@ func GetTlsConfigurationForServer(cstream s2av2pb.S2AService_SetUpSessionClient,
...
-	// tls.ClientAuthType = tls.NoClientCert, tls.RequestClientCert or
-	// tls.RequireAnyClientCert. For these 3 ClientAuthType's, go crypto/tls
-	// skips normal verification, and directly calls VerifyPeerCertificate func.
-	// (2) Get ClientCAs field from S2Av2, if S2Av2 gives tls.ClientAuthType =
-	// tls.VerifyClientCertIfGiven or tls.RequireAndVerifyClientCert.
-	// For these 2 ClientAuthType's, go crypto/tls performs normal verification,
-	// which requires ClientCAs, before custom VerifyPeerCertificate func.
-	clientCApool := x509.NewCertPool()
-	clientCApool.AppendCertsFromPEM(clientCert)
+	clientAuth := getTLSClientAuthType(tlsConfig)
```
[the remark this answers is not in the document]
Comment (rmehta19) [reply internal/v2/tls_config_store/tls_config_store.go:171]: Yes. I manually tested on this branch.
fakes2av2 already sets ```RequestClientCertificate``` to ```s2av2pb.GetTlsConfigurationResp_ServerTlsConfiguration_REQUEST_AND_REQUIRE_CLIENT_CERTIFICATE_AND_VERIFY``` which maps to ```tls.RequireAnyClientCert```

In fakes2av2.go, I triggered an error in ```verifyClientPeer``` by changing line 248 from :
```
        if ok := rootCertPool.AppendCertsFromPEM(clientCert); ok != true {
```
to : 
```
        if ok := rootCertPool.AppendCertsFromPEM(clientCert); ok == true {
```
From this change, I expect to see this error(line 249 in fakes2av2.go):
```
                "Client Peer Verification failed: S2Av2 could not obtain/parse roots."
```
I then ran the existing E2E test in s2av2_e2e_test.go and got this output:
image 

Also, s2a_e2e_test.go results in this:
image

Both these tests show the handshake terminating and retrying but getting the same error.

This confirms that tls.RequireAnyClientCert skips normal verification and goes straight to our ```ValidatePeerCertificate``` and terminates the handshake if this function throws an error.

[the remark this answers is not in the document]
Comment (matthewstevenson88) [reply internal/v2/tls_config_store/tls_config_store.go:171]: Perfect, thanks for confirming!

**Review by matthewstevenson88 — APPROVED** [review]

**Pull request merged**